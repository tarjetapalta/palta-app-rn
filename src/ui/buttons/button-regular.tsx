import React, { FunctionComponent, useContext } from 'react';
import {
  ViewProps,
  TouchableOpacityProps,
  ActivityIndicator,
} from 'react-native';

import { KeyColors } from '@theme/colors';
import styled, { ThemeContext } from '@theme/styled-components';
import {
  WrapperButton,
  TextButton,
  TouchableOpacity,
  TextButtonProps,
} from './base';

const Wrapper = styled(WrapperButton)<{ color?: KeyColors }>`
  ${({ theme, color }) =>
    color &&
    `
    background-color: ${theme.colors[color]};
  `}
`;

type ButtonRegularType = Pick<ViewProps, 'testID'>;

interface ButtonRegularProps
  extends ButtonRegularType,
    Pick<TouchableOpacityProps, 'accessibilityLabel' | 'disabled'>,
    TextButtonProps {
  text: string;
  onActionHandle: () => void;
  loading?: boolean;
  backgroundColor?: KeyColors;
}

export const ButtonRegular: FunctionComponent<ButtonRegularProps> = ({
  /** Test id for test */
  testID,
  /** Text to be displayed inside button. */
  text,
  /** Action function. */
  onActionHandle,
  /** VoiceOver know what element they have selected */
  accessibilityLabel,
  /** Toucheable disabled */
  disabled,
  /** If true, disable all interactions for this component and replace the title with an ActivityIndicator. */
  loading,
  /** Background color */
  backgroundColor,
  /** Text color */
  color,
}) => {
  const theme = useContext(ThemeContext);

  return (
    <TouchableOpacity
      testID={testID}
      onPress={onActionHandle}
      accessibilityLabel={accessibilityLabel}
      disabled={disabled || loading}
    >
      <Wrapper disabled={disabled} color={backgroundColor}>
        {loading ? (
          <ActivityIndicator color={theme.colors.white} />
        ) : (
          <TextButton color={color}>{text}</TextButton>
        )}
      </Wrapper>
    </TouchableOpacity>
  );
};
