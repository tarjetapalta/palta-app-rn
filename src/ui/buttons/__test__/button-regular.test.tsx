import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { ButtonRegular } from '../button-regular';

describe('buttonRegular component', () => {
  it('basic button regular with text, onActionHandle, accessibilityLabel and testID "ButtonRegular"', () => {
    const result = renderer.create(
      <WithTheme>
        <ButtonRegular
          text="Button test"
          onActionHandle={() => null}
          accessibilityLabel="button test"
        />
      </WithTheme>,
    );

    expect(result).toMatchSnapshot();
  });
  it('basic button regular with disabled"', () => {
    const result = renderer.create(
      <WithTheme>
        <ButtonRegular
          text="Button test"
          onActionHandle={() => null}
          accessibilityLabel="button test"
          disabled
        />
      </WithTheme>,
    );

    expect(result).toMatchSnapshot();
  });
  it('basic button regular with loading"', () => {
    const result = renderer.create(
      <WithTheme>
        <ButtonRegular
          text="Button test"
          onActionHandle={() => null}
          accessibilityLabel="button test"
          loading
        />
      </WithTheme>,
    );

    expect(result).toMatchSnapshot();
  });
  it('basic button regular with custom testID"', () => {
    const result = renderer.create(
      <WithTheme>
        <ButtonRegular
          testID="custom testID"
          text="Button test"
          onActionHandle={() => null}
          accessibilityLabel="button test"
        />
      </WithTheme>,
    );

    expect(result).toMatchSnapshot();
  });
});
