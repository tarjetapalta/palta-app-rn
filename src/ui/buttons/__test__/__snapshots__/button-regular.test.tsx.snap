// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonRegular component basic button regular with custom testID" 1`] = `
<View
  accessibilityLabel="button test"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="custom testID"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#333333",
          "borderRadius": 2,
          "borderTopColor": "#272727",
          "borderTopWidth": 4,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "paddingBottom": 2,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "BwModelica-Bold",
            "fontSize": 14,
            "letterSpacing": 0.21,
            "opacity": 0.87,
            "textAlign": "center",
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Button test
    </Text>
  </View>
</View>
`;

exports[`buttonRegular component basic button regular with disabled" 1`] = `
<View
  accessibilityLabel="button test"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    disabled={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#333333",
          "borderRadius": 2,
          "borderTopColor": "#272727",
          "borderTopWidth": 4,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 0.38,
          "paddingBottom": 2,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "BwModelica-Bold",
            "fontSize": 14,
            "letterSpacing": 0.21,
            "opacity": 0.87,
            "textAlign": "center",
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Button test
    </Text>
  </View>
</View>
`;

exports[`buttonRegular component basic button regular with loading" 1`] = `
<View
  accessibilityLabel="button test"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#333333",
          "borderRadius": 2,
          "borderTopColor": "#272727",
          "borderTopWidth": 4,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "paddingBottom": 2,
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#ffffff"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;

exports[`buttonRegular component basic button regular with text, onActionHandle, accessibilityLabel and testID "ButtonRegular" 1`] = `
<View
  accessibilityLabel="button test"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#333333",
          "borderRadius": 2,
          "borderTopColor": "#272727",
          "borderTopWidth": 4,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "paddingBottom": 2,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "BwModelica-Bold",
            "fontSize": 14,
            "letterSpacing": 0.21,
            "opacity": 0.87,
            "textAlign": "center",
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Button test
    </Text>
  </View>
</View>
`;
